client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -249,14 +214,25 @@ public void fillBuffer() {
             }
         } catch (SegmentTruncatedException e) {
             log.warn("Encountered exception filling buffer", e);
+            return Futures.failedFuture(e);
         }
+        return outstandingRequest == null ? CompletableFuture.completedFuture(null) : outstandingRequest.thenApply(r -> null);
     }
     
     @Override
     @Synchronized
-    public boolean isSegmentReady() {
-        boolean result = receivedEndOfSegment || receivedTruncated || buffer.dataAvailable() > 0 || (outstandingRequest != null && outstandingRequest.isDone());
-        log.trace("isSegmentReady {} on segment {} status is {}", result, getSegmentId(), this);
+    public int bytesInBuffer() {
+        int result = buffer.dataAvailable();
+        boolean atEnd = receivedEndOfSegment || receivedTruncated || (outstandingRequest != null && outstandingRequest.isCompletedExceptionally());
```

Comment:
This changes the behavior of `io.pravega.client.stream.impl.Orderer#nextSegment`.
Earlier the Orderer would choose a segment which had data in buffer or has reached the end (either of receivedEndOfSegment, recievedTruncated). Is this by design?